tap_dashboard_reports/streams.py

```diff
@@ -31,10 +33,13 @@ def name(self):
 
     @property
     def primary_keys(self):
-        """Return primary key dynamically based on user inputs."""
-        return self._report.get(
-            "key_properties", self._dimensions
-        )  # or self._dimensions
+        """Return primary key dynamically based on user inputs.
+        If in config doesn't have key key_property then take _dimensions
+        """
+        if self.custom_key:
+            return self._report["key_property"]
```

Comment:
As `self._dimensions` is a list, I think it should looks like that:
```
return [self._report["key_property"]]
```

Reply:
change logic to return list